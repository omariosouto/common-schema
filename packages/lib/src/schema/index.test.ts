import { describe, expect, it } from "vitest";
import { parseSchema, s } from "./index";

describe("Wire In", () => {
  it("wire in schemas must always be loose and accept random data, but when parsed returns only what is expected", () => {
    const data = {
      name: "John Doe",
      age: 30,
      extraField: "This should be ignored",
    };

    const schema = s.looseObject({
      name: s.string(),
      age: s.number(),
    });

    const parsed = parseSchema(schema, data);

    console.log("Parsed data:", parsed);

    expect(parsed).toEqual({
      name: "John Doe",
      age: 30,
      // The extra field is ignored by the typing
      extraField: "This should be ignored",
    });
  });
});

describe("Wire Out", () => {
  it("wire out schemas must always be strict and reject random data, but when parsed returns only what is expected", () => {
    const data = {
      name: "John Doe",
      age: 30,
      extraField: "This should be ignored",
    };

    const schema = s.strictObject({
      name: s.string(),
      age: s.number(),
    });

    expect(() => parseSchema(schema, data)).toThrowErrorMatchingSnapshot();
  });
});