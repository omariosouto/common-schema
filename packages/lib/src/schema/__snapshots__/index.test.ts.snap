// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Wire Out > wire out schemas must always be strict and reject random data, but when parsed returns only what is expected 1`] = `
[SchemaError: [
  {
    "code": "unrecognized_keys",
    "keys": [
      "extraField"
    ],
    "path": [],
    "message": "Unrecognized key: \\"extraField\\""
  }
]]
`;
